src/execution_plan/ops/op_sort.c

```diff
@@ -239,10 +197,21 @@ void SortFree(OpBase *ctx) {
 		op->buffer = NULL;
 	}
 
-	if(op->expressions) {
-		for(int i = 0; i < array_len(op->expressions); i++) AR_EXP_Free(op->expressions[i]);
-		array_free(op->expressions);
-		op->expressions = NULL;
+	if(op->exps) {
+		uint exp_count = array_len(op->exps);
+		for(uint i = 0; i < exp_count; i ++) {
+			// The parser does not generate identifiers for ORDER BY expressions,
+			// so the resolved_name is constructed by us and must be freed.
```

Comment:
Oh, I want this very much! It's not trivial currently because ORDER BY nodes are of type `CYPHER_AST_SORT_ITEM`, which don't have associated identifiers (they're actually fairly different from RETURN projects). Hopefully the annotation PR will allow us to perform this kind of augmentation, not sure yet!